src/app/src/components/Sidebar.js

```diff
@@ -52,14 +48,44 @@ export default function Sidebar() {
 }
 
 function ReferenceLayers() {
-    const dispatch = useDispatch();
-    const id = useBoundaryId();
-    const addReferenceImage = useAddReferenceImage(id);
-    const [postReferenceImage, { error }] = useUpdateReferenceImageMutation();
-    useEndpointToastError(error);
-    const openFileDialog = useFilePicker(files => files.map(addReferenceImage));
+    const boundaryId = useBoundaryId();
+
+    const {
+        data: boundary,
+        isLoading,
+        error,
+    } = useGetBoundaryDetailsQuery(boundaryId);
+
+    const [createReferenceImage, { createReferenceImageError }] =
+        useUploadReferenceImageMutation();
+
+    const [updateReferenceImage, { updateReferenceImageError }] =
+        useDebouncedUpdateReferenceImageMutation(boundaryId);
+
+    useEndpointToastError(
+        createReferenceImageError ?? updateReferenceImageError
+    );
+
+    const uploadImage = file => {
+        createReferenceImage({
+            boundaryId,
+            filename: file.name,
+            is_visible: true,
+            distortion: null,
+            opacity: 100,
+        });
+    };
+
+    // TODO support multiple files
```

Comment:
This might work as is, but I'm worried about manual cache updates and race conditions. I think we have two approaches here:
- Process uploads synchronously
- Update the api to support batch upload